### PR TITLE
Several fixes to processing of cross-sections

### LINKED ISSUE #145: Improve support for xyz cross-sections
### Version checks

- [X] I have checked that this issue has not already been reported.
- [X] I have checked that this bug exists on the latest version.

### Reproducible Example

xyz-cross-sections can be assigned. However, a check for consistency of the cross-section is missing. As per definition, a cross-section should consist of at least 3 points.

### Current behaviour

Invalid xyz-cross sections can be generated by the code

### Desired behaviour

Improve checks to remove invalid xyz-cross-sections

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shartgring**
## Issue addressed
Fixes #145 #146 and adds new functionality

## Explanation
I have included the code related to cross-sections from Santa Cruz branch, and also some other additional functionalities and fixes.

## Checklist
- [X] Updated tests or added new tests
- [X] Branch is up to date with `main`
- [X] Tests & pre-commit hooks pass
- [X] Updated documentation if needed
- [X] Updated changelog.rst if needed

## Additional Notes (optional)
I also copied some code for 'special' cross-sections from Santa Cruz; I am not sure what that refers to to be honest. Maybe leave that out / make an issue for additional functionality where we temporary store the code?

The tests still pass, I had to include ``reset_index`` which I did not particularly liked as the local test build failed due to a pandas error (index and column both had name crs_id).

I also spend quite some time on a test for one of the cross-section workflows. I think we can use it as a template to test the other cross-section types when we have (hopefully) more time in the future to tidy things up.

### CODE REVIEW

**Comment by shartgring** [comment]
I have included the code related to cross-sections from Santa Cruz branch, and also some other additional functionalities and fixes.

I also copied some code for 'special' cross-sections from Santa Cruz; I am not sure what that refers to to be honest. Maybe leave that out / make an issue for additional functionality where we temporary store the code?

The tests still pass, I had to include `reset_index` which I did not particularly liked as the local test build failed due to a pandas error (index and column both had name crs_id).

**Comment by shartgring** [comment]
Ai I see a test is failing now. Maybe because I added the ``maxdist`` variable last-minute. Will explore and fix it

**Comment by shartgring** [comment]
Testing works again and tests passed succesfully :)

File: hydromt_delft3dfm/dflowfm.py [hunk 1]
```diff
@@ -1287,9 +1304,25 @@ def _setup_crosssections(
             # set crsloc and crsdef attributes to crosssections
             self.logger.info(f"Preparing 1D point crossections from {crosssections_fn}")
             gdf_cs = workflows.set_point_crosssections(
-                branches, gdf_cs, maxdist=self._network_snap_offset
+                branches, gdf_cs, maxdist=maxdist
+            )
+        elif crosssections_type == "special":  # read from the output
```
Comment (hboisgon) [inline hydromt_delft3dfm/dflowfm.py:1309]: I thought cross-sections type could only be "branch", "xyz", "point". What is special? SHould it be added to the docstrings of the different functions?
Additionally could this part be tested?

**Review by hboisgon — APPROVED** [review]
Thanks for the fixes @shartgring !
I guess the model now runs in delft3d fm too? Just one question for cross-sections type special but the rest looks good.
Thanks for adding a test already. Codecov seems to think it's not enough though, do you think you can easily increase coverage on the parts you added or leave it for later for now?

File: hydromt_delft3dfm/dflowfm.py (hunk 1 above)
Comment (shartgring) [reply hydromt_delft3dfm/dflowfm.py:1309]: I checked with Xiaohan and the 'special' x-section can indeed be removed

**Pull request merged**